bundle/scripts/scripts.go

```diff
@@ -0,0 +1,68 @@
+package scripts
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/config"
+	"github.com/databricks/cli/libs/cmdio"
+)
+
+func Execute(hook config.ScriptHook) bundle.Mutator {
+	return &script{
+		scriptHook: hook,
+	}
+}
+
+type script struct {
+	scriptHook config.ScriptHook
+}
+
+func (m *script) Name() string {
+	return fmt.Sprintf("scripts.%s", m.scriptHook)
+}
+
+func (m *script) Apply(ctx context.Context, b *bundle.Bundle) error {
+	out, err := executeHook(ctx, b, m.scriptHook)
+	cmdio.LogString(ctx, bytes.NewBuffer(out).String())
+	return err
+}
+
+func executeHook(ctx context.Context, b *bundle.Bundle, hook config.ScriptHook) ([]byte, error) {
+	command := getCommmand(b, hook)
+	if command == "" {
+		return nil, nil
+	}
+	commands := strings.Split(strings.ReplaceAll(string(command), "\r\n", "\n"), "\n")
+
+	out := make([][]byte, 0)
+	for _, command := range commands {
+		if command == "" {
+			continue
+		}
+
+		subcommands := strings.Split(command, " && ")
```

Comment:
List of commands is supported (see example of config in PR description) in a form of multiline string. This helps to get a better UX for configuring because users might want to provide only one or multiple commands so supporting only list of commands is not very user friendly. 

As to `&&`: it works on Windows as well and it's purely for again simplifying UX, but I don't have a very strong preference if we want to support this

Reply:
This way of "lexing" the command ourselves is pretty neat! This allows us to force stronger cross-platform compatibility across scripts. I'd love to try this in "experimental" and see how well it works compared to invoking an OS-specific shell.

Reply:
This may break if e.g. someone writes `sh -c 'a && b && c'`. But hopefully this is an edge case?

Reply:
I'm not too keen on lexing. Too many edge cases. I took a peek at NPM and it passes the script string to `sh` verbatim ([doc](https://docs.npmjs.com/cli/v9/using-npm/scripts#exiting)). What are we solving by doing the lexing ourselves?

Reply:
For experimental support, what we have here is fine. The trade-off here is around portability. The syntax supported by `sh` is not portable. The proposed implementation here is much more narrow and would be more portable. But for the final version I think we should just go back to `sh`/`$SHELL`/`cmd.exe`. Likely with some strict checks or warnings to help portability.

Reply:
Moved to using `sh` in this commit https://github.com/databricks/cli/pull/632/commits/17d4472bd9515d59fd6670a8866b9a9e9a6db0e1